File: freecodecamp/daily-code-challenge/2026-03-08.js
function isValidHSL(hsl) {
  const regex = /^hsl\(\s*(\d+)\s*,\s*(\d+)%\s*,\s*(\d+)%\s*\)\s*;?\s*$/;

  const match = hsl.match(regex);
  if (!match) return false;

  const h = Number(match[1]);
  const s = Number(match[2]);
  const l = Number(match[3]);

  return h >= 0 && h <= 360 &&
         s >= 0 && s <= 100 &&
         l >= 0 && l <= 100;
}

File: freecodecamp/daily-code-challenge/2026-03-08.py
import re

def is_valid_hsl(hsl):
    pattern = r'^hsl\(\s*(\d+)\s*,\s*(\d+)%\s*,\s*(\d+)%\s*\)\s*;?\s*$'
    
    match = re.match(pattern, hsl)
    if not match:
        return False
    
    h = int(match.group(1))
    s = int(match.group(2))
    l = int(match.group(3))
    
    return 0 <= h <= 360 and 0 <= s <= 100 and 0 <= l <= 100
